src/H5Fint.c

```diff
@@ -1029,19 +1029,22 @@ H5F_prefix_open_file(bool try, H5F_t **_file, H5F_t *primary_file, H5F_prefix_op
  *
  * Purpose:     Check the file signature to detect an HDF5 file.
  *
- * Return:      true/false/FAIL
+ * Return:      SUCCEED/FAIL
  *-------------------------------------------------------------------------
  */
-htri_t
-H5F__is_hdf5(const char *name, hid_t fapl_id)
+herr_t
+H5F__is_hdf5(const char *name, hid_t fapl_id, bool *is_hdf5)
 {
     H5FD_t       *lf        = NULL;        /* Low-level file struct            */
     H5F_shared_t *shared    = NULL;        /* Shared part of file              */
     haddr_t       sig_addr  = HADDR_UNDEF; /* Address of hdf5 file signature    */
-    htri_t        ret_value = FAIL;        /* Return value                     */
+    herr_t        ret_value = SUCCEED;     /* Return value                     */
 
     FUNC_ENTER_PACKAGE
 
+    /* Reset output parameter */
+    *is_hdf5 = false;
```

Comment:
This would potentially modify the argument even in the case of failure.

Reply:
Yes, I considered that, but the previous behavior was aggregating FAIL with 'false' and it seemed best to continue that.

Reply:
OK, I'll update the code.